main/src/library/Vector.flix

```diff
@@ -0,0 +1,1389 @@
+/*
+ *  Copyright 2023 Magnus Madsen, Stephen Tetley
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// A Vector data type which is fundamentally like Array but is immutable.
+///
+
+type alias Vector[a] = Array[a, true]
```

Comment:
We would want instances here, but we cannot add those until `Vector` becomes a type in its own right.